src/main/psxsdk/libsnd/vmanager.c

```diff
@@ -31,7 +31,115 @@ INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SeAutoPan);
 
 INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SetAutoPan);
 
-INCLUDE_ASM("main/nonmatchings/psxsdk/libsnd/vmanager", SpuVmInit);
+struct _ss_spu_vm_rec_struct {
+    u32 pad[2];
+    u32 D_8003BD50;
+};
+
+extern struct _ss_spu_vm_rec_struct _ss_spu_vm_rec;
+
+extern s16 _svm_vcf;
+extern s16 _svm_orev1;
+extern s16 _svm_orev2;
+extern s8 _svm_auto_kof_mode;
+
+void SpuVmInit(u8 arg0) {
+    u16 temp_v1;
+    u16 var_a1;
+    s32 var_v1;
+    u16* temp_a0;
+    s32 var_a0;
+    s32 bitsUpper;
+    s32 bitsLower;
+    u16 temp;
+    _spu_setInTransfer(0);
+    _svm_vcf = 0;
+    _svm_damper = 0;
+    SpuInitMalloc(0x20, &_ss_spu_vm_rec.D_8003BD50);
+    for (var_a1 = 0; var_a1 < 8 * 24; var_a1++) {
+        _svm_sreg_buf.raw[var_a1] = 0;
+    }
+    for (var_a1 = 0; var_a1 < 24; var_a1++) {
```

Comment:
```suggestion
    for (i = 0; i < NUM_SPU_CHANNELS; i++) {
```
and it would be good to use `NUM_SPU_CHANNELS` with the other instances of `24`